### PR TITLE
Update memory.md

### PR DESCRIPTION
**Pull request by wilzbach**
Idea: the current text is hard to understand for someone not familiar with function annotation and @safe.

Previously: https://github.com/stonemaster/dlang-tour/pull/456

### CODE REVIEW

**Comment by ZombineDev** [comment]
BTW, I'm working on full rewrite of this page. I was meaning to open an issue to share my ideas, but I ran out of time this week. I hope to get back to it next week.

In short, I think that we should move the memory chapter later in the tour and make it more detailed and more appealing for system programmers (in addition to the beginners, this tour aims for). Also, the `@safe`-ty attributes should be moved earlier, while an expanded section about low-level stuff should be expanded later.

In the meantime, this LGTM.

**Comment by ZombineDev** [comment]
Also, we would need to revisit safety w.r.t. `scope` if/when DIP1000 is implemented.

File: basics/memory.md [hunk 1]
```diff
@@ -20,18 +20,18 @@ As soon as the memory referenced by `a` isn't referenced anymore
 through any variable in the program, the garbage collector
 will free its memory.
 
-D also allows pointer arithmetic, except in
-code that is marked as @safe.
+D has three different security levels for functions: `@system`, `@trusted`, and `@safe`.
```
Comment (ZombineDev) [inline basics/memory.md]: I'm not sure if "security" is right word, but I'll leave it for now, as overall this PR is an improvement.

Comment (ZombineDev) [inline basics/memory.md]: (Security may have slightly different meaning to Java/.NET and web developers)

Comment (wilzbach) [inline basics/memory.md]: > I'm not sure if "security" is right word, but I'll leave it for now, as overall this PR is an improvement.

I agree, but I couldn't find a better word when I submitted the PR -> #3

**Comment by wilzbach** [comment]
> BTW, I'm working on full rewrite of this page. I was meaning to open an issue to share my ideas, but I ran out of time this week. I hope to get back to it next week.

Looking forward :)

> In short, I think that we should move the memory chapter later in the tour and make it more detailed and more appealing for system programmers (in addition to the beginners, this tour aims for). Also, the @safe-ty attributes should be moved earlier, while an expanded section about low-level stuff should be expanded later.

Yep I absolutely agree (I think I even opened an issue once, but I can't find it atm).

> Also, we would need to revisit safety w.r.t. scope if/when DIP1000 is implemented.

I don't hope it, but it will take a while until this lands in stable (earliest date is 2.073, which is ~Nov 2016), so we are "safe" for now  ;-)

**Pull request merged**